### PR TITLE
feat: add html_head_file to inject custom head scripts into the app

### PR DESCRIPTION
**Pull request by mscolnick**
You can include a custom HTML head file to add additional functionality to your notebook, such as analytics, custom fonts, meta tags, or external scripts. The contents of this file will be injected into the `<head>` section of your notebook.

To include a custom HTML head file, specify the relative file path in your app configuration. This can be done through the marimo editor UI in the notebook settings (top-right corner).

Some examples:
1. **Google Analytics**
2. **Custom Fonts**
3. **Meta Tags**
4. **External Scripts and Libraries**

### CODE REVIEW

File: frontend/src/components/app-config/app-config-form.tsx [hunk 1]
```diff
@@ -146,6 +146,34 @@ export const AppConfigForm: React.FC = () => {
...
+                    }}
+                  />
+                </FormControl>
+                <FormMessage />
+              </FormItem>
+              <FormDescription>
+                A filepath to an HTML file to be injected into the{" "}
+                <Kbd className="inline">{"<head/>"}</Kbd> section of the
+                notebook. Use this to add analytics, custom fonts, meta tags, or
+                external scripts.
```
Comment (mscolnick) [inline frontend/src/components/app-config/app-config-form.tsx:172]: maybe too wordy?

**Review by akshayka — APPROVED** [review]
Nice. I like how this one change has many use cases.
